src/hooks/useFirstMainDocument.ts

```diff
@@ -13,7 +13,8 @@ import { RWAccess } from '@tdev-models/helpers/accessPolicy';
  */
 export const useFirstMainDocument = <Type extends DocumentType>(
     documentRootId: string | undefined,
-    meta: TypeMeta<Type>
+    meta: TypeMeta<Type>,
+    createDocument: boolean = true
```

Comment:
If a component wraps a document-creating component (such as `SelfCheckTaskState`, which wraps `TaskState`), and that wrapper also makes a call to `useFirstDocument()`, multiple documents may be created for the same documentRoot. This can lead to unexpected behavior. This parameter allows us to prevent document creation in components that are not the "primary responsible component" for that document type.

Another option might be to change `DocumentStore.ts#create` to use the `documentRoot` in the abort signal, rather than a random UUID. However, this might also have undesired side effects.